### PR TITLE
Simplify EM_ASM handling

### PR DESCRIPTION
**Pull request by kripken**
Emscripten counterpart to WebAssembly/binaryen#3044 , see background there,
and https://github.com/WebAssembly/binaryen/issues/3043

This basically gets rid of the special handling of EM_ASMs. Instead, they are just
normal JS library methods like any other.

(The one interesting thing in them, unchanged from before this PR, is that we
need to read the varargs before doing any proxying, see the comment there for
more details.)

### CODE REVIEW

File: tests/browser/test_em_asm_blocking.cpp [hunk 1]
```diff
@@ -7,8 +8,15 @@ std::atomic<int> ret;
...
     var elem = document.getElementById('elem');
+    window.almost_PI = 3.14159;
     return elem.innerText.length;
   });
+  double almost_PI = MAIN_THREAD_EM_ASM_DOUBLE({
+    // read a double from the main thread
+    return window.almost_PI;
+  });
+  printf("almost PI: %f\n", almost_PI);
+  assert(fabs(almost_PI - 3.14159) < 0.001);
```
Comment (sbc100) [inline tests/browser/test_em_asm_blocking.cpp:19]: What the changes to the test code?

File: src/library.js [hunk 2]
```diff
@@ -4186,12 +4229,6 @@ LibraryManager.library = {
   llvm_dbg_value: function() {},
   llvm_debugtrap: function() {},
   llvm_ctlz_i32: function() {},
-  emscripten_asm_const: function() {},
```
Comment (sbc100) [inline src/library.js:4189]: This function no longer exists?   I guess it was never needed?

File: tests/browser/test_em_asm_blocking.cpp (hunk 1 above)
Comment (kripken) [reply tests/browser/test_em_asm_blocking.cpp:19]: Just some added testing. We tested `MAIN_THREAD_EM_ASM_INT` but not `_DOUBLE`.

File: src/library.js (hunk 2 above)
Comment (kripken) [reply src/library.js:4189]: Yes, it may have made sense in some old approach to EM_ASM in fastcomp perhaps.

**Pull request merged**